src/framework/components/render/component.js

```diff
@@ -0,0 +1,823 @@
+                    batcher.insert(BatchGroup.RENDER, value, this.entity);
+                }
+
+                if (value < 0 && this._batchGroupId >= 0 && this.enabled && this.entity.enabled) {
+                    // re-add render to scene, in case it was removed by batching
+                    this.addToLayers();
+                }
+
+                this._batchGroupId = value;
+            }
+        }
+    },
+
+    "material": {
+        get: function () {
+            return this._material;
+        },
+        set: function (value) {
+            if (this._material !== value) {
+                this._material = value;
+
+                if (this._meshInstances && this._type !== 'asset') {
+                    for (var i = 0; i < this._meshInstances.length; i++) {
+                        this._meshInstances[i].material = value;
+                    }
+                }
+            }
+        }
+    },
+
+    "materialAssets": {
+        get: function () {
+            return this._materialReferences.map(function (ref) {
+                return ref.id;
+            });
+        },
+
+        set: function (value) {
+            var i;
+            value = value || [];
+            if (this._materialReferences.length > value.length) {
+                for (i = value.length; i < this._materialReferences.length; i++) {
+                    this._materialReferences[i].id = null;
+                }
+                this._materialReferences.length = value.length;
+            }
+
+            for (i = 0; i < value.length; i++) {
+                if (!this._materialReferences[i]) {
+                    this._materialReferences.push(
+                        new AssetReference(
+                            i,
+                            this,
+                            this.system.app.assets, {
+                                add: this._onMaterialAdded,
+                                load: this._onMaterialLoad,
+                                remove: this._onMaterialRemove,
+                                unload: this._onMaterialUnload
+                            },
+                            this
+                        )
+                    );
+                }
+
+                if (value[i]) {
+                    var id = value[i] instanceof Asset ? value[i].id : value[i];
+                    if (this._materialReferences[i].id !== id) {
+                        this._materialReferences[i].id = id;
+                    }
+
+                    if (this._materialReferences[i].asset) {
+                        this._onMaterialAdded(i, this, this._materialReferences[i].asset);
+                    }
+                } else {
+                    this._materialReferences[i].id = null;
+
+                    if (this._meshInstances[i]) {
+                        this._meshInstances[i].material = this.system.defaultMaterial;
+                    }
+                }
+            }
+        }
+    },
+
+    "asset": {
```

Comment:
might have been useful to support multiple assets given that `Entity` only supports a single render component.

Reply:
but an asset represents an array of meshInstances/materials .. so that's probably enough?

Reply:
I'm thinking of the case where a user wants to add a cigar (render component) to a hand node in the hierarchy, when the node already has a render component attached. User would have to create a child node and attach the cigar there, which actually is probably fine.

Reply:
yep that's how I'd expect them to do it .. either that, or dont smoke.